File: modules/angular2/src/core/facade/exceptions.dart
library angular.core.facade.exceptions;

import 'exception_handler.dart';
export 'exception_handler.dart';

class BaseException extends Error {
  final String message;

  BaseException([this.message]);

  String toString() {
    return this.message;
  }
}

class WrappedException extends Error {
  final dynamic context;
  final String wrapperMessage;
  final originalException;
  final originalStack;

  WrappedException(
      [this.wrapperMessage, this.originalException, this.originalStack, this.context]);

  get message { return ExceptionHandler.exceptionToString(this); }

  String toString() { return this.message; }
}

Error makeTypeError([String message = ""]) {
  return new BaseException(message);
}

dynamic unimplemented() {
  throw new BaseException('unimplemented');
}

File: es6/prod/src/facade/exceptions.js
import { ExceptionHandler } from './exception_handler';
export { ExceptionHandler } from './exception_handler';
export class BaseException extends Error {
    constructor(message = "--") {
        super(message);
        this.message = message;
        this.stack = (new Error(message)).stack;
    }
    toString() { return this.message; }
}
/**
 * Wraps an exception and provides additional context or information.
 */
export class WrappedException extends Error {
    constructor(_wrapperMessage, _originalException, _originalStack, _context) {
        super(_wrapperMessage);
        this._wrapperMessage = _wrapperMessage;
        this._originalException = _originalException;
        this._originalStack = _originalStack;
        this._context = _context;
        this._wrapperStack = (new Error(_wrapperMessage)).stack;
    }
    get wrapperMessage() { return this._wrapperMessage; }
    get wrapperStack() { return this._wrapperStack; }
    get originalException() { return this._originalException; }
    get originalStack() { return this._originalStack; }
    get context() { return this._context; }
    get message() { return ExceptionHandler.exceptionToString(this); }
    toString() { return this.message; }
}
export function makeTypeError(message) {
    return new TypeError(message);
}
export function unimplemented() {
    throw new BaseException('unimplemented');
}
